### ISSUE TITLE
Subplot does not inherit file name from gmtbegin

### ISSUE THREAD

**Issue by adigitoleo** [GenericMappingTools/GMT.jl#937]
A slight pain point from the combination of `gmtbegin` and `subplot`:

Julia 1.7.3
GMT (system) 6.3
GMT.jl 0.42.4 (also on master)

```julia
julia> gmtbegin("example.png")

julia> subplot(grid = (2, 2), dims = (10, 10))

julia> plot(sin.(1:10), axes = (; fill = :lightgreen), grid = true, panel = 4)

julia> gmtend()
```

The file is saved using the default name `GMTplot.png` instead of `example.png`.

**Comment by joa-quim** [comment]
There are still shits to be solved when using both ``gmtbegin(...) ... subplot(...) ...`` but meanwhile this works
```
subplot(grid = (2, 2), dims = (10, 10), figname="example.png")
    plot(sin.(1:10), axes = (; fill = :lightgreen), grid = true, panel = 4)
subplot(:end)
```

**Comment by adigitoleo** [comment]
Ah OK it is `figname`, I was trying `filename` haha. Thanks for the workaround, that's fine for now.

**Comment by joa-quim** [comment]
The original file name issue plus the ``subplot-inside-a-gmtbegin-end`` should have been solved by #940

**Comment by adigitoleo** [comment]
Thanks, both the implicit way (just `subplot` -> `subplot(:end)`) and explicit way (inside `gmtbegin/gmtend`) are now working.
